cranelift/jit/src/memory.rs

```diff
@@ -113,15 +113,17 @@ pub(crate) struct Memory {
     already_protected: usize,
     current: PtrLen,
     position: usize,
+    enable_branch_protection: bool,
 }
 
 impl Memory {
-    pub(crate) fn new() -> Self {
+    pub(crate) fn new(enable_branch_protection: bool) -> Self {
```

Comment:
Rather than take a single `bool` here, can we take an options struct? It may seem overkill for one flag but `enable_branch_protection` would not be my first guess for the meaning of a bare `false` hanging out in a `Memory::new` call :-)

Reply:
I decided to define a new `enum` and pass it instead. I noticed that Windows defined some memory protection flags that were related to Control Flow Guard, its own CFI implementation, and this approach seems like a good way to leave the possibility of adding support for them in the future (if we decide to do so), while keeping the current implementation simple.